File: Python3/Find Bottom Left Tree Value.py
#1
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        queue = [root]
        while queue:
            node = queue.pop(0)
            res = node.val
            if node.right:
                queue.append(node.right)
            if node.left:
                queue.append(node.left)
        return res

#2
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        q  = collections.deque()
        q.append(root)
        res = 0
        while q:
            node = q.popleft()
            res = node.val            
            if node.right:
                q.append(node.right)

            if node.left:
                q.append(node.left)        
        return res

#3
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        queue = [root]
        for node in queue:
            queue += filter(None, (node.right, node.left))
        return node.val

File: Golang/Find Bottom Left Tree Value.go
//1
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findBottomLeftValue(root *TreeNode) int {
    s := []*TreeNode{root}
    var res int
    for s != nil {
        res = s[0].Val
        s = do(s)
    }
    return res
    
}


func do(s []*TreeNode) []*TreeNode {
    var res []*TreeNode
    for _, v := range s {
        if v.Left != nil {
            res = append(res, v.Left)
        }
        if v.Right != nil {
            res = append(res, v.Right)
        }
    }
    return res
}

//2
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func findBottomLeftValue(root *TreeNode) int {
    que := []*TreeNode{root}

    var popped *TreeNode

    for len(que) > 0 {
    popped = que[len(que) - 1]
    que = que[:len(que) - 1]

    if popped.Right != nil {
        que = append([]*TreeNode{popped.Right}, que...)
    }

    if popped.Left != nil {
        que = append([]*TreeNode{popped.Left}, que...)
    }
    }

    return popped.Val
}